app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -113,14 +115,35 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
       },
     });
 
+    const mergedScopes = mergeScopes(
+      flattenedRequiredScopes,
+      flattenedOptionalScopes,
+    );
+
+    // clear per scope middleware
+    hooks.multichainMiddlewareManager.removeAllMiddleware();
+    hooks.multichainSubscriptionManager.unsubscribeAll();
```

Comment:
would this clear the middleware for active connections on other origins?

Reply:
💯 ➕